### ISSUE TITLE
etcd_backup unexpected EOF

### ISSUE THREAD

**Issue by drpebcak** [rancher/terraform-provider-rancher2-archive#88]
When using the new `rancher2_etcd_backup` resource, I get this error during a `terraform apply`:

```
* module.example.rancher2_etcd_backup.cluster: 1 error(s) occurred:

* rancher2_etcd_backup.cluster: unexpected EOF
```

My resource looks like this:
```
resource "rancher2_etcd_backup" "cluster" {
  backup_config {
    enabled        = true
    interval_hours = 6
    retention      = 21

    s3_backup_config {
      access_key  = "${var.etcd_backup_iam_access_key}"
      bucket_name = "${var.etcd_backup_bucket}"
      endpoint    = "s3.${var.aws_region}.amazonaws.com"
      region      = "${var.aws_region}"
      secret_key  = "${var.etcd_backup_iam_secret_key}"
    }
  }

  cluster_id = "${rancher2_cluster.rke.id}"
  name       = "${var.etcd_backup_name}"
}
```

Running it a second time gave this error:
```
* rancher2_etcd_backup.cluster: Bad response statusCode [409]. Status [409 Conflict]. Body: [baseType=error, code=AlreadyExists, message=etcdbackups.management.cattle.io "example-backup-name" already exists] from [https://example.example.rancher.io/v3/etcdbackups]
```

**Comment by drpebcak** [comment]
Looking at that backup config in the api, it does not have my s3 settings in it.

**Comment by drpebcak** [comment]
@rawmind0 any ideas on this one?

**Comment by rawmind0** [comment]
@drpebcak, it seems that `S3_backup_config` is not saved through API, on etcd backups for rke clusters if not configured first on rke template. Working well with imported clusters.

Please, try to configure S3_backup_config on rke cluster first and you could configure `etcd_backup` with S3 config.

```
resource "rancher2_cluster" "foo" {
  name = "foo"
  description = "Terraform custom cluster acceptance test"
  rke_config {
    network {
      plugin = "canal"
    }
    services {
      etcd {
        creation = "12h"
        retention = "72h"
        backup_config {
          enabled = true
          interval_hours = 20
          retention = 10
          s3_backup_config {
            access_key = "access_key"
            bucket_name = "bucket_name"
            endpoint = "endpoint"
            region = "region"
            secret_key = "secret_key"
          }
        }
      }
    }
  }
}
```

I'm going to investigate why is happening, if it's a issue on API.

**Comment by drpebcak** [comment]
This seems like a step in the right direction, however now it shows a modification every time I try to apply the terraform:

```
  ~ module.example.rancher2_cluster.rke
      rke_config.0.services.0.etcd.0.backup_config.0.s3_backup_config.0.secret_key: <sensitive> => <sensitive> (attribute changed)
```

**Comment by rawmind0** [comment]
Added fix on PR #90

**Comment by drpebcak** [comment]
#90 fixes my original issue as well as the problem I saw with the workaround.
